src/sources/closed_coil.cpp

```diff
@@ -0,0 +1,634 @@
+    for (auto f : el_faces)
+      pushIfUnique(wedge_faces, f);
+  }
+
+  for (auto wf : wedge_faces) {
+
+    int e1, e2;
+    mesh_parent_->GetFaceElements(wf, &e1, &e2);
+
+    // If the face is a coil boundary
+    if (!(isInDomain(e1, coil_domains_, mesh_parent_) &&
+          isInDomain(e2, coil_domains_, mesh_parent_))) {
+      continue;
+    }
+
+    // If the face is not true interior
+    if (!(mesh_parent_->FaceIsInterior(wf) ||
+          (mesh_parent_->GetFaceInformation(wf).tag ==
+               mfem::Mesh::FaceInfoTag::SharedConforming ||
+           mesh_parent_->GetFaceInformation(wf).tag ==
+               mfem::Mesh::FaceInfoTag::SharedSlaveNonconforming))) {
+      continue;
+    }
+
+    // If the face is shared between two elements internal to the wedge
+    bool test1 = false;
+    bool test2 = false;
+    for (auto e : wedge_els) {
+      if (e == e1)
+        test1 = true;
+      if (e == e2)
+        test2 = true;
+    }
+
+    if (test1 && test2)
+      continue;
+
+    // If the face is part of the first electrode
+    test1 = false;
+    for (auto b_fc : bdr_els) {
+      if (wf == mesh_parent_->GetBdrFace(b_fc)) {
+        test1 = true;
+        break;
+      }
+    }
+    if (test1)
+      continue;
+
+    // At last, if the face is none of these things, it must be our second
+    // electrode
+    auto *new_elem = mesh_parent_->GetFace(wf)->Duplicate(mesh_parent_);
+    new_elem->SetAttribute(elec_attrs_.second);
+    mesh_parent_->AddBdrElement(new_elem);
+  }
+
+  // Only after this do we set the domain attributes
+  for (auto e : wedge_els)
+    mesh_parent_->SetAttribute(e, new_domain_attr_);
+
+  mesh_parent_->FinalizeTopology();
+  mesh_parent_->Finalize();
+  mesh_parent_->SetAttributes();
+}
+
+void ClosedCoilSolver::initChildMeshes() {
+
+  mfem::Array<int> doms_array;
+  SubdomainToArray(coil_domains_, doms_array);
+  mesh_[0] = new mfem::ParSubMesh(
+      mfem::ParSubMesh::CreateFromDomain(*mesh_parent_, doms_array));
+
+  hephaestus::Subdomain wedge("wedge", new_domain_attr_);
+  SubdomainToArray(wedge, doms_array);
+  mesh_[1] = new mfem::ParSubMesh(
+      mfem::ParSubMesh::CreateFromDomain(*mesh_parent_, doms_array));
+
+  for (int i = 0; i < 2; ++i) {
+    inheritBdrAttributes(mesh_parent_, mesh_[i]);
+  }
+}
+
+void ClosedCoilSolver::makeFESpaces() {
+
+  for (int i = 0; i < 2; ++i) {
+    H1_Collection_[i] =
+        new mfem::H1_FECollection(order_, mesh_[i]->Dimension());
+    HCurl_Collection_[i] =
+        new mfem::ND_FECollection(order_, mesh_[i]->Dimension());
+    H1FESpace_[i] =
+        new mfem::ParFiniteElementSpace(mesh_[i], H1_Collection_[i]);
+    HCurlFESpace_[i] =
+        new mfem::ParFiniteElementSpace(mesh_[i], HCurl_Collection_[i]);
+  }
+}
+
+void ClosedCoilSolver::makeGridFunctions() {
+
+  for (int i = 0; i < 2; ++i) {
+    V_[i] = new mfem::ParGridFunction(H1FESpace_[i]);
+    J_[i] = new mfem::ParGridFunction(HCurlFESpace_[i]);
```

Comment:
Check if V_[i] is being overwritten?

Reply:
Done, although depending on how often Init() is used by the executioner, I might need to do similar checks for several other members of the class